### PR TITLE
Added initial part of Directory

### PR DESCRIPTION
**Pull request by marcserrat**

### CODE REVIEW

File: src/ConnectClient.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+/**
+ * This file is part of the Ingram Micro Cloud Blue Connect SDK.
+ *
+ * @copyright (c) 2018. Ingram Micro. All Rights Reserved.
+ */
+
+namespace Connect;
+
+class ConnectClient extends AutomationEngine
```
Comment (vgrebenschikov) [inline src/ConnectClient.php]: why extending Engine?

Comment (marcserrat) [reply src/ConnectClient.php]: ConnectClient is intented to be a new entrypoint for everything. AutomationEngine allows the mounting of all modules and is abstract. thanks to this you can do things like:
$connect = new Connect\ConnectClient();
$connect->directory->getAssetById('AS-463-716-763-4')->getRequests();
$connect->fulfillment->listRequests();
$connect->usage->getUsageTemplateFile();

Overall idea is to start for a future major release force people to have 1 app entry point

File: src/AutomationEngine.php [hunk 2]
```diff
@@ -99,4 +104,18 @@ public function __call($name, $arguments)
...
+    /**
+     * @param Config|null $configuration
+     * @param Container|null $container
+     * @return AutomationEngine
+     * @throws ConfigException
+     */
+    public static function getInstance(Config $configuration = null, Container $container = null)
+    {
+        if (!(self::$_instance instanceof self)) {
+            new static($configuration, $container);
```
Comment (vgrebenschikov) [inline src/AutomationEngine.php]: return new static(...)

Comment (marcserrat) [reply src/AutomationEngine.php]: is returned just after. If is not initialized it creates it.

**Pull request merged**